pkgs/by-name/co/code-cursor/package.nix

```
@@ -54,15 +66,29 @@ let
     runHook postInstall
   '';
 
+  darwinInstallPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/Applications
+    cp -r *.app $out/Applications/
+
+    mkdir -p "$out/bin"
+    cat << EOF > "$out/bin/cursor"
+    open -na '$APP_DIR' --args "\$@"
+    EOF
+    chmod +x "$out/bin/cursor"
+    runHook postInstall
+  '';
+
 in
 stdenvNoCC.mkDerivation {
   inherit pname version;
 
-  src = wrappedAppImage;
+  inherit src;
```

```suggestion
  src = if stdenvNoCC.isLinux then wrappedAppImage else source;
```

Linux needs the wrapped app image. Otherwise, we select the dmg.